src/core/loader/cia.h

```diff
@@ -0,0 +1,367 @@
+  unsigned char padding3[4];
+  unsigned char twlflag;
+  unsigned char padding4[0x31];
+  unsigned char accessrights[4];
+  unsigned char titleversion[2];
+  unsigned char contentcount[2];
+  unsigned char bootcontent[2];
+  unsigned char padding5[2];
+  unsigned char hash[32];
+  unsigned char contentinfo[36*64];
+} ctr_tmd_body;
+
+typedef struct
+{
+  unsigned char index[2];
+  unsigned char commandcount[2];
+  unsigned char unk[32];
+} ctr_tmd_contentinfo;
+
+
+typedef struct
+{
+  unsigned char id[4];
+  unsigned char index[2];
+  unsigned char type[2];
+  unsigned char size[8];
+  unsigned char hash[32];
+} ctr_tmd_contentchunk;
+
+
+typedef struct
+{
+  unsigned char signaturetype[4];
+  unsigned char signature[256];
+} ctr_tmd_header_2048;
+
+typedef struct
+{
+  unsigned char signaturetype[4];
+  unsigned char signature[512];
+} ctr_tmd_header_4096;
+
+
+typedef struct
+{
+  u8 enable_timelimit[4];
+  u8 timelimit_seconds[4];
+} timelimit_entry;
+
+typedef struct
+{
+  u8 sig_type[4];
+  u8 signature[0x100];
+  u8 padding1[0x3c];
+  u8 issuer[0x40];
+  u8 ecdsa[0x3c];
+  u8 padding2[0x03];
+  u8 encrypted_title_key[0x10];
+  u8 unknown;
+  u8 ticket_id[8];
+  u8 console_id[4];
+  std::array<u8,8> title_id;
+  u8 sys_access[2];
+  u8 ticket_version[2];
+  u8 time_mask[4];
+  u8 permit_mask[4];
+  u8 title_export;
+  u8 commonkey_idx;
+  u8 unknown_buf[0x30];
+  u8 content_permissions[0x40];
+  u8 padding0[2];
+
+  timelimit_entry timelimits[8];
+} eticket;
+
+typedef enum
+{
+  KEY_ERR_LEN_MISMATCH,
+  KEY_ERR_INVALID_NODE,
+  KEY_OK
+} keystatus;
+
+typedef enum
+{
+  RSAKEY_INVALID,
+  RSAKEY_PRIV,
+  RSAKEY_PUB
+} rsakeytype;
+
+typedef struct
+{
+  unsigned char n[256];
+  unsigned char e[3];
+  unsigned char d[256];
+  unsigned char p[128];
+  unsigned char q[128];
+  unsigned char dp[128];
+  unsigned char dq[128];
+  unsigned char qp[128];
+  rsakeytype keytype;
+} rsakey2048;
+
+typedef struct
+{
+  unsigned char data[16];
+  int valid;
+} key128;
+
+typedef struct
+{
+  key128 commonkey;
+  key128 titlekey;
+  key128 ncchkey;
+  key128 ncchfixedsystemkey;
+  rsakey2048 ncsdrsakey;
+  rsakey2048 ncchrsakey;
+  rsakey2048 ncchdescrsakey;
+  rsakey2048 firmrsakey;
+} keyset;
+
+typedef struct
+{
+  char pathname[MAX_PATH];
+  int valid;
+} filepath;
+
+typedef struct
+{
+  keyset keys;
+  filepath exefspath;
+  filepath exefsdirpath;
+  filepath firmdirpath;
+  filepath romfspath;
+  filepath romfsdirpath;
+  filepath exheaderpath;
+  filepath logopath;
+  filepath plainrgnpath;
+  filepath certspath;
+  filepath contentpath;
+  filepath tikpath;
+  filepath tmdpath;
+  filepath metapath;
+  filepath lzsspath;
+  filepath wavpath;
+  unsigned int mediaunitsize;
+  int ignoreprogramid;
+  int listromfs;
+  u32 cwavloopcount;
+} settings;
+
+struct tik_context
+{
+  u64 offset;
+  u32 size;
+  u8 titlekey[16];
+  eticket tik;
+  ctr_aes_context aes;
+  settings* usersettings;
+};
+
+typedef struct
```

Comment:
for C++ you wouldn't use a typedef - check other instances in this file.
Check other files in Citra too, maybe we also do this in other spots - it's fine in that case.